packages/zotero.rb

```diff
@@ -0,0 +1,127 @@
+    system "git clone https://github.com/zotero/translators.git"
+    system "git clone https://github.com/zotero/bundled-styles.git"
+    system "git clone https://github.com/zotero/pdf-worker.git"
+    system "git clone https://github.com/zotero/note-editor.git"
+    system "git clone https://github.com/zotero/pdf-reader.git"
+    system "git clone https://github.com/zotero/zotero-schema.git"
+    system "git clone https://github.com/gildas-lormeau/SingleFile.git"
+    system "git clone https://github.com/zotero/utilities.git"
+    system "git clone https://github.com/zotero/translate.git"
+    system "git clone https://github.com/citation-style-language/locales.git"
+    system "git clone https://github.com/egh/zotero-transfw.git"
+    system "git clone https://github.com/zotero/zotero-libreoffice-integration.git"
+    system "git clone https://github.com/zotero/pdf.js.git"
+
+    system "echo downloaded resources"
+
+    system "ls -l -a"
+    Dir.chdir("zotero-build") {
+      system "git submodule init"
+      system "echo \"#{$srcdir}/zotero-transfw\""
+      system "git config submodule.xpi/zotero-transfw.url \"#{$srcdir}/zotero-transfw\""
+      system "git -c protocol.file.allow=always submodule update"
+      system "echo 'Step 1: PASSED'"
+    }
+    
+    Dir.chdir("zotero") {
+      system "git submodule init"
+      system "git config submodule.translators.url '#{$srcdir}/translators'"
+      system "git config submodule.styles.url '#{$srcdir}/bundled-styles'"
+      system "git config submodule.pdf-worker.url '#{$srcdir}/pdf-worker'"
+      system "git config submodule.note-editor.url '#{$srcdir}/note-editor'"
+      system "git config submodule.pdf-reader.url '#{$srcdir}/pdf-reader'"
+      system "git config submodule.resource/schema/global.url '#{$srcdir}/zotero-schema'"
+      system "git config submodule.resource/SingleFile.url '#{$srcdir}/SingleFile'"
+      system "git config submodule.chrome/content/zotero/xpcom/utilities.url '#{$srcdir}/utilities'"
+      system "git config submodule.chrome/content/zotero/xpcom/translate.url '#{$srcdir}/translate'"
+      system "git config submodule.chrome/content/zotero/locale/csl.url '#{$srcdir}/locales'"
+      system "git -c protocol.file.allow=always submodule update"
+      system "echo 'Step 2: PASSED'"
+
+      Dir.chdir("pdf-reader") {
+        system "git submodule init"
+        system "git config submodule.pdf.js.url '#{$srcdir}/pdf.js'"
+        system "git -c protocol.file.allow=always submodule update"
+      }
+
+      Dir.chdir("pdf-worker") {
+        system "git submodule init"
+        system "git config submodule.pdf.js.url '#{$srcdir}/pdf.js'"
+      }
+
+      system "echo 'Installing LTS node.js...'"
+
+      system "nodebrew install v18.16.1 || true" # if node version is already present or locale is not correct somehow it'd throw an error and exit the script
+      system "nodebrew exec v18.16.1 npm i --legacy-peer-deps"
```

Comment:
You don't want to do this. You want to tell the user to install a particular version of nodebrew if necessary.
Then you can just use a variable to run the program:
```
      system "nodebrew exec #{@nodebrew_version} npm i --legacy-peer-deps"
```

Reply:
See https://github.com/chromebrew/chromebrew/pull/8493/files#r1271123996